### PR TITLE
rustscan 1.9.0 (new formula)

### PR DESCRIPTION
**Pull request by bee-san**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/libsndfile.rb [hunk 1]
```diff
@@ -12,9 +12,9 @@ class Libsndfile < Formula
 
   bottle do
     cellar :any
-    sha256 "8e192a4b76ad637395dfce818e2eec62e3a90722b33759fefcd4af389498bb9c" => :catalina
-    sha256 "1784457959d0c8ca513223f12e6f884f0dafe74ea0e9511e8c7d1b2557cfbb91" => :mojave
-    sha256 "3ae061cb03f627870e9fe98cd65f538547f109f472d20d235103ab4625d22b94" => :high_sierra
+    sha256 "bf93a0083eecbcadc7eb391b9f429e0f55c4f571b5f4d9104cfaf60f8e5932dc" => :catalina
+    sha256 "ef1bd6eb6f6d33e22a05116b0893d85422e3ad7fced4252eace97232e523075d" => :mojave
+    sha256 "9d7d184ccf41eff27b8425a8fd50d60fa8a383338efcd0dcf7121f0ce7e3545c" => :high_sierra
```
Comment (Moisan) [inline Formula/libsndfile.rb:17]: Those changes are not related to rustscan, please remove them.

File: Formula/rustscan.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  depends_on "rust" => :build
+  depends_on "nmap"
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    system "rustscan", "--greppable", "127.0.0.1"
+    system "rustscan", "--greppable", "0.0.0.0"
+    assert_equal 1, 1
```
Comment (Moisan) [inline Formula/rustscan.rb:17]: This `assert_equal` doesn't test anything, please remove it.

**Review by Moisan — CHANGES_REQUESTED** [review]
Thank you for you first PR @bee-san. A few changes are needed, please squash your commits after updating your branch. Also please change your commit message to `rustscan 1.9.0 (new formula)`.

File: Formula/rustscan.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+class Rustscan < Formula
+  desc "Modern Day Portscanner"
+  homepage "https://github.com/rustscan/rustscan"
+  url "https://github.com/RustScan/RustScan/archive/1.9.0.tar.gz"
+
```
Comment (dawidd6) [inline Formula/rustscan.rb:5]:
```suggestion
```

File: Formula/rustscan.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  sha256 "9923b91310be342f5fcec42796662b53f4d0b05ec5339c44b801959ebfe824dc"
+  license "GPL-3.0-or-later"
+  depends_on "rust" => :build
```
Comment (dawidd6) [inline Formula/rustscan.rb:8]:
```suggestion

   depends_on "rust" => :build
```

File: Formula/rustscan.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  depends_on "nmap"
+  def install
```
Comment (dawidd6) [inline Formula/rustscan.rb:10]:
```suggestion

  def install
```

File: Formula/rustscan.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    system "rustscan", "--greppable", "127.0.0.1"
```
Comment (dawidd6) [inline Formula/rustscan.rb:15]:
```suggestion
    system bin/"rustscan", "--greppable", "127.0.0.1"
```

File: Formula/rustscan.rb [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    system "rustscan", "--greppable", "0.0.0.0"
```
Comment (dawidd6) [inline Formula/rustscan.rb:16]:
```suggestion
    system bin/"rustscan", "--greppable", "0.0.0.0"
```

**Comment by dawidd6** [comment]
Name your commit as the PR title.

**Comment by bee-san** [comment]
Hey, the suggestion to use bin/rustscan instead of `rustscan` makes the CI fail, specifically the tests fail:

```
An exception occurred within a child process:
  BuildError: Failed executing: bin/rustscan --greppable 127.0.0.1
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:2026:in `block in system'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1964:in `open'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1964:in `system'
/home/bee/Documents/Projects/rustscan/homebrew-core/Formula/rustscan.rb:16:in `block in <class:Rustscan>'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1835:in `block (3 levels) in run_test'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/utils.rb:496:in `with_env'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1834:in `block (2 levels) in run_test'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:894:in `with_logging'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1833:in `block in run_test'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/mktemp.rb:57:in `block in run'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/mktemp.rb:57:in `chdir'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/mktemp.rb:57:in `run'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:2075:in `mktemp'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1827:in `run_test'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/test.rb:42:in `block in <main>'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/timeout.rb:93:in `block in timeout'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/timeout.rb:33:in `block in catch'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/timeout.rb:33:in `catch'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/timeout.rb:33:in `catch'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/timeout.rb:108:in `timeout'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/test.rb:41:in `<main>'
```

If I `brew install --build-from-source` and run the tests by hand, they work. I don't understand why they are failing with `brew test rustscan.rb`, any suggestions? :smile:

**Comment by jonchang** [comment]
Should be `"#{bin}/rustscan"` in the test block I'm assuming?

**Comment by bee-san** [comment]
> Should be `"#{bin}/rustscan"` in the test block I'm assuming?

Thanks a lot! That' solved it. :)

File: Formula/rustscan.rb [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+  depends_on "rust" => :build
+  depends_on "nmap"
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    system "#{bin}/rustscan", "--greppable", "127.0.0.1"
+    system "#{bin}/rustscan", "--greppable", "0.0.0.0"
```
Comment (SMillerDev) [inline Formula/rustscan.rb:17]: Can this be checked for output? Something like this
```suggestion
    assert_match "error", shell_output("#{bin}/rustscan --greppable 127.0.0.1")
    assert_match "error", shell_output("#{bin}/rustscan --greppable 0.0.0.0")
```

Comment (bee-san) [reply Formula/rustscan.rb:17]: Hey! This assumes that RustScan, when it errors, prints only the word "error".

I'm trying to figure out how to do `assert_no_match` and check to see if the result contains the world "panic" (which is an error that is guaranteed to be in all RustScan error messages), but my Ruby skills are lackluster :sweat_smile:

**Review by dawidd6 — APPROVED** [review]

**Review by Moisan — APPROVED** [review]

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@bee-san, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!
